server/setup_flow.go

```diff
@@ -93,6 +85,26 @@ func (p *Plugin) NewSetupFlow() *flow.Flow {
 		InitHTTP(p.router)
 }
 
+func (p *Plugin) NewOAuth2Flow() *flow.Flow {
+	pluginURL := *p.client.Configuration.GetConfig().ServiceSettings.SiteURL + "/" + "plugins" + "/" + manifest.ID
+	conf := p.getConfig()
+	return flow.NewFlow("setup-oauth2", p.client, pluginURL, conf.botUserID).
+		WithSteps(
+			p.stepCloudOAuthConfigure(),
+			p.stepInstalledJiraApp(),
+			p.stepWebhook(),
+			p.stepWebhookDone(),
+			p.stepConnect(),
+			p.stepConnected(),
+			p.stepAnnouncementQuestion(),
+			p.stepAnnouncementConfirmation(),
+			p.stepCancel(),
+			p.stepDone(),
+		).
```

Comment:
Interesting approach. So this is sort of a copy of the setup flow, but assumes the admin is using Cloud OAuth?

Reply:
The main purpose of using this approach is because the existing setup flow might or might not have triggered. Since we are reusing the code from the setup_flow when a user directly runs the slash command `/jira instance install cloud-oauth <url>`, we are starting the oauth2Flow for the user instead of initiating the setup flow from an intermediate step from the `NewSetupFlow`. I have tried with the later case and it does not seems to be working at all cases, i.e user might/might not have triggered the setup flow before.